### PR TITLE
Applying latest changes on aieng-template

### PR DESCRIPTION
**Pull request by lotif**
# PR Type
Feature | Documentation

# Short Description

Clickup Ticket(s): https://app.clickup.com/t/8687bp131

Applying latest changes from [aieng-template](https://github.com/VectorInstitute/aieng-template), which includes Ruff for python checks, pip-audit and documentation.

# Tests Added
NA

### CODE REVIEW

File: .pre-commit-config.yaml
Comment (amrit110) [inline .pre-commit-config.yaml:47]: change to florist

File: docs/source/api.rst
Comment (amrit110) [inline docs/source/api.rst:8]: would need to be updated to florist modules

File: docs/source/conf.py
Comment (amrit110) [inline docs/source/conf.py:18]: change to florist package path

File: docs/source/conf.py
Comment (amrit110) [inline docs/source/conf.py:59]: change to florist package path

File: docs/source/reference/api/aieng_template.bar.rst
Comment (amrit110) [inline docs/source/reference/api/aieng_template.bar.rst:1]: the files under reference/api can be removed. When you run the `make html` command after specifying the top level rst files, it will be automatically generated

File: docs/source/user_guide.md
Comment (amrit110) [inline docs/source/user_guide.md:1]: please update to florist user guide material

**Comment by lotif** [comment]
thanks for the help, @amrit110 !

**Comment by lotif** [comment]
@amrit110 made all the changes, please check again.

@jewelltaylor this PR turned out to be a lot bigger than I initially planned but I thought it would be more productive to just integrate the latest changes from the aieng template, which includes pip-audit and also Ruff. I will merge this only after you merge your other PR to avoid crazy conflicts with your poetry.lock  😅

**Review by amrit110 — APPROVED** [review]

**Pull request merged**
